pkg/daemon/ceph/osd/daemon.go

```diff
@@ -349,12 +349,24 @@ func getAvailableDevices(context *clusterd.Context, desiredDevices []DesiredDevi
 					// the desired devices is a regular expression
 					matched, err = regexp.Match(desiredDevice.Name, []byte(device.Name))
 					if err != nil {
-						logger.Errorf("regex failed on device %s and filter %s. %+v", device.Name, desiredDevice.Name, err)
+						logger.Errorf("regex failed on device %q and filter %q. %+v", device.Name, desiredDevice.Name, err)
 						continue
 					}
-					logger.Infof("device %s matches device filter %s: %t", device.Name, desiredDevice.Name, matched)
+					logger.Debugf("device %q matches device filter %q: %t", device.Name, desiredDevice.Name, matched)
```

Comment:
This log is recorded even if the filter is not matched, per device and per filter.
So we will see many log lines like `device sda matches device filter ^sd[bc]: false`.
These logs give the information of "which filter caused the device to be selected,"
but I think this type of information is for debugging.

It would be informative to log the final result of filtering, so I will add logging of
"device ... is selected" at the INFO level after processing all filters.

Reply:
I would tend to `Info` logging here. Contextually, this log appears during OSD provisioning and not in the main operator, so I like to consider what information is likely to be helpful in 80% of cases where users are looking at this log. I imagine that many users might want to validate their selection of device filter in the logs, and debug logging is not on by default. I also don't think this will add overly much logging as `Info` that it will be a detriment to the user.

Reply:
Aha, surely, it's important to provide helpful logs to most users.
I got extra help from these types of logs indeed when I was less experienced in Rook.

I will rearrange the logs as follows.
- keep logging of matching steps as `Infof`
- enrich the log of `device %q is selected`, which I add in this PR, with the information of the related filter
- output the summary of matched and non-matched devices after the loop